components/range_cache_memory_engine/src/range_stats.rs

```diff
@@ -381,15 +337,139 @@ impl RangeStatsManager {
         };
         (regions_to_load, regions_to_evict)
     }
+
+    // Evict regions with less read flow until the memory usage is below the soft
+    // limit threshold.
+    pub(crate) async fn evict_on_soft_limit_reached<F>(
+        &self,
+        mut evict_region: F,
+        delete_range_scheduler: &Scheduler<BackgroundTask>,
+        cached_region_ids: Vec<u64>,
+        memory_controller: &MemoryController,
+    ) where
+        F: FnMut(
+            &CacheRegion,
+            EvictReason,
+            Option<Box<dyn AsyncFnOnce + Send + Sync>>,
+        ) -> Vec<CacheRegion>,
+    {
+        // Get regions' stat of the cached region and sort them by next + prev in
+        // descending order.
+        let regions_activity = self
+            .info_provider
+            .get_regions_stat(cached_region_ids.clone())
+            .unwrap();
+        if regions_activity.is_empty() {
+            return;
+        }
+
+        // Use the average mvcc amplification to filter the regions with high mvcc
+        // amplification
+        let avg_mvcc_amplification = regions_activity
+            .iter()
+            .map(|(_, ra)| ra.cop_detail.mvcc_amplification())
+            .sum::<f64>()
+            / regions_activity.len() as f64;
+
+        let evict_candidates: Vec<_> = {
+            let mut regions_loaded = self.region_loaded_at.write().unwrap();
+            let mut evict_candidates = regions_activity;
+            info!(
+                "IME evict candidate count before filter";
```

Comment:
```suggestion
                "ime evict candidate count before filter";
```

Let's the log style consistent.